### PR TITLE
Fix date time field display issue

### PR DESCRIPTION
**Pull request by julienWebmecanik**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.3.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Some datetime are not displayed according to user local timezone : 
- in contact details
- in contact audit logs details

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. create a new customfield of type datetime for a contact
Screenshot 2021-03-02 at 15 24 14

3. edit a contact and set the field
Screenshot 2021-03-02 at 15 28 41

4. see that your customfield is displayed in accordance with your local timezone (mine is Europe/Paris) in contact details and in auditlog 
Screenshot 2021-03-02 at 15 37 10

Screenshot 2021-03-02 at 15 38 47

### CODE REVIEW

**Comment by RCheesley** [comment]
Thanks for making this PR @julienWebmecanik and welcome to the community!

@cla-bot check

**Comment by npracht** [comment]
@julienWebmecanik can your PR fix also https://github.com/mautic/mautic/issues/9616 ?

**Comment by npracht** [comment]
Please have a look at tests issue also ;)

**Comment by julienWebmecanik** [comment]
> @julienWebmecanik can your PR fix also #9616 ?

Tried to reproduce unsuccessfully, are we sure this is still an issue ?

**Review by npracht — COMMENTED** [review]
👍🏽 works properly :)

**Review by npracht — APPROVED** [review]

**Comment by RCheesley** [comment]
@julienWebmecanik I was trying to reproduce this issue before applying the PR but I cannot.

Here is a screencast: https://youtu.be/ASiEWawyaNU

I followed your steps with the default timezone and then changed it to Paris per your example (and cleared the cache just to be sure!) but the date/time is still showing as it was entered.

If I am missing something here do please point out any missing steps!

**Comment by julienWebmecanik** [comment]
@RCheesley sorry for the response time.
The reason you don't see an issue is because you edited the field before changing your timezone (when you were in UTC).
If you edit your field again after having changed your timezone it should produce the bug.

The issue is as follows : 
- When a custom datetime field is saved, the conversion ```local_timezone``` -> ```utc``` is done (as all datetimes are stored utc). (so the field DateTime in user/edit is acting properly, both ways)
- when dates are retrieved and displayed in contact details, the conversion ```utc``` ->```local_timezone``` is not done, thus rendering a "wrong" datetime.

I hope this makes it clearer.

File: app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php [hunk 1]
```diff
@@ -110,4 +110,12 @@ function (string $key, array $parameters = []) {
 
         $this->assertSame('00:00:00', $this->helper->toText($dateTime));
     }
+
+    public function testFullConcat()
+    {
+        date_default_timezone_set('Europe/Paris');
```
Comment (tejasync) [inline app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: Setting time zone this way, will set it for other test cases as well.

**Review by tejasync — CHANGES_REQUESTED** [review]

**Comment by RCheesley** [comment]
@julienWebmecanik please take a look at the comment from @ts-navghane about the test.

File: app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: I suppose timezone is reset to old on each test by tearDownAfterClass

https://github.com/mautic/mautic/pull/9747/files/dec7d4456ef73d76d25482632bea72a8b57fbbd6#diff-94502daac727de55e303995909aa3441edad6e827d5e027bc1ee84ac9114910bR45

`date_default_timezone_set` is used in other 4 tests in that class.


Don't see issue.

**Comment by kuzmany** [comment]
@RCheesley https://github.com/mautic/mautic/pull/9747#discussion_r615383106 

For me it's OK.

File: app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php (hunk 1 above)
Comment (tejasync) [reply app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: Sorry, I forgot to add the example test cases earlier which I had written to prove what I meant.

Here is the screenshot of an example:
Screenshot 2021-04-19 at 13 57 07

In test case 1 (testDefaultTimeZone), I checked for the default timezone set (which was UTC).
After this I set it to Europe/Paris.

In test case 2 (testSetTimeZoneInOtherTestCase), I again checked for the default timezone and it was Europe/Paris.

As you can see, both the test cases pass.

Comment (tejasync) [reply app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: Also, the updated time zone after it is set, won't affect test cases in another Test.php file though.
If this is okay, I do not see an issue.

Comment (kuzmany) [reply app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: I understand, but If you check https://github.com/mautic/mautic/blob/features/app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php

date_default_timezone_set is apply in each method


And https://github.com/mautic/mautic/blob/features/app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php#L45

set timezone back to default after all classes method passes

Comment (tejasync) [reply app/bundles/CoreBundle/Tests/Unit/Templating/Helper/DateHelperTest.php:116]: Yes, understood!
I see no issues. 👍🏻

**Review by tejasync — APPROVED** [review]
I have tested this and see no further issues.

**Pull request merged**